CMakeLists.txt

```diff
@@ -57,19 +130,17 @@ endif()
 
 set(CMAKE_CXX_STANDARD 17)
 
-set(CMAKE_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/include)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib)
-set(CMAKE_INSTALL_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib)
-
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/lib")
-
 add_subdirectory(apps)
 add_subdirectory(core)
-add_subdirectory(extern)
 add_subdirectory(plugins)
 add_subdirectory(io)
-add_subdirectory(tests)
+
+# Set up the test(s).
+include( CTest )
+if( BUILD_TESTING AND DETRAY_BUILD_TESTING )
```

Comment:
Is this also true if only DETRAY_BENCHMARKS is set or is that a dependent option on DETRAY_BUILD_TESTING now? Because "benchmark" is a subdir to the "test" dir right now. Maybe I am just confused

Reply:
What will be the long-term future of those benchmarks? I do consider them a type of test. In fact, I intend to stop them from being installed together with detray in the future.

So I was going with the "all benchmarks are tests, but not all tests are benchmarks" approach on setting up these flags. :thinking: In this setup, if `DETRAY_BUILD_TESTING==FALSE` (or even more generally `BUILD_TESTING==FALSE`), benchmarks would not be built. Independent of the value of `DETRAY_BUILD_BENCHMARKS`. This latter flag would only get a say in what's happening, if "tests in general" are being built.

Reply:
No that's exactly what I thought. The becnhmarks and later the consistency checks will probably take a long time and should not be put on the CI. I was just confused, because it is an option instead of a dependent option. But maybe a dependent option should ont be used?